hive/src/test/java/org/apache/iceberg/hive/HiveCreateReplaceTableTest.java

```diff
@@ -155,7 +155,7 @@ public void testReplaceTableTxn() {
     txn.commitTransaction();
 
     Table table = catalog.loadTable(TABLE_IDENTIFIER);
-    Assert.assertEquals("Partition spec should match", PartitionSpec.unpartitioned(), table.spec());
+    Assert.assertEquals("Partition spec should be unpartitioned", 0, table.spec().fields().size());
```

Comment:
Is any `PartitionSpec`with `0` fields the `unpartitioned` spec or should it match a certain spec (like in the removed version)? Seems like the `equals` method takes `fields` and `specId` into the contract.

Reply:
Any partition spec with 0 fields is considered unpartitioned.

Reply:
I should note that this needed to be updated because the spec's ID didn't match, causing the test to fail.

`PartitionSpec.unpartitioned()` doesn't necessarily have the right ID for a given table. We primarily use it when creating tables, where the spec gets rebuilt and assigned the right ID.

Reply:
Yeah, I supposed that was the reason for the change (spec's ID not matching). Thanks for the context!